extensions/amp-story/0.1/amp-story.js

```diff
@@ -502,12 +520,49 @@ export class AmpStory extends AMP.BaseElement {
   }
 
 
+  /** @private */
+  waitForInitialLoad_() {
+    const pagesToWaitFor = this.isDesktop_()
+      ? [this.pages_[0], this.pages_[1]] : [this.pages_[0]];
+
+    return this.waitForPageLoadOrTimeout_(pagesToWaitFor, PAGE_LOAD_TIMEOUT_MS);
+  }
+
+  /**
+   * @param {!Array<!./amp-story-page.AmpStoryPage>} pagesToWaitFor The page(s)
+   *     to wait for.
+   * @param {number=} opt_timeoutMs The maximum amount of time to wait, in
+   *     milliseconds.  Waits indefinitely if unspecified.
+   * @return {!Promise} A promise that is resolved when the page is loaded or
+   *     the timeout has been exceeded, whichever happens first.
+   * @private
+   */
+  waitForPageLoadOrTimeout_(pagesToWaitFor, opt_timeoutMs) {
+    const loadPromise = Promise.all(
+        pagesToWaitFor.map(page => page.waitUntilLoaded()));
+
+    if (opt_timeoutMs === undefined) {
+      return loadPromise;
+    }
+
+    return Promise.race([
+      loadPromise,
+      this.timer_.promise(opt_timeoutMs),
+    ]);
+  }
+
+  /** @private */
+  markStoryAsLoaded_() {
+    dispatch(this.element, EventType.STORY_LOADED, true);
```

Comment:
Is there a listener for this?

Reply:
Not within the runtime; it's meant for embedding pages to listen to (e.g. test environments, tools, etc)

Reply:
sg.